src/main/java/com/cburch/logisim/gui/start/Startup.java

```diff
@@ -566,74 +432,165 @@ private static RC handleArgGeometry(Startup startup, Option opt) {
       h = Integer.parseInt(wxh[1]);
     } catch (NumberFormatException e) {
       logger.error(S.get("argGeometryError"));
-      return RC.QUIT;
+      return RC.CLI_ERROR;
     }
     if (w <= 0 || h <= 0) {
       logger.error(S.get("argGeometryError"));
-      return RC.QUIT;
-    }
-    AppPreferences.WINDOW_WIDTH.set(w);
-    AppPreferences.WINDOW_HEIGHT.set(h);
-    if (loc != null) {
-      AppPreferences.WINDOW_LOCATION.set(x + "," + y);
+      return RC.CLI_ERROR;
     }
+    windowSize = new Point(w,h);
+    if (loc != null) windowLocation = new Point(x,y);
     return RC.OK;
   }
 
-  private static RC handleArgLocale(Startup startup, Option opt) {
-    setLocale(opt.getValue());
+  private RC handleArgLocale(Option opt) {
+    String lang = opt.getValue();
+    final var opts = S.getLocaleOptions();
+    for (final var locale : opts) {
+      if (lang.equals(locale.toString())) {
+        LocaleManager.setLocale(locale);
+        return RC.OK;
+      }
+    }
+    logger.error(S.get("invalidLocaleError"));
+    logger.error(S.get("invalidLocaleOptionsHeader"));
+    for (final var o : opts) {
```

Comment:
Please no single letter vars for new code.

Reply:
Personally, I don't have problems with single letter names for local variables - especially if the context in which they are used/valid is small. I agree that argument names, member variables, constants, and other identifiers, which are visible in larger scopes should have longer meaningful names. E.g., I see on object `S` used all over the place in this file and others of this PR, which is IMHO not good practice.

Reply:
S is used all over, not just in this file. I do not intend to change that--we can create a new issue for that very global change.
I will find a non-single-letter var name.